engine/common/rpc/convert/validate_test.go

```diff
@@ -0,0 +1,303 @@
+
+		// This address should be invalid for Testnet chain
+		_, err := convert.Address(rawAddr, chain)
+		require.Error(t, err)
+		assert.Equal(t, codes.InvalidArgument, status.Code(err))
+		assert.Contains(t, err.Error(), "is invalid for chain")
+	})
+}
+
+// TestValidateHexToAddress tests the HexToAddress validation function.
+func TestValidateHexToAddress(t *testing.T) {
+	t.Parallel()
+
+	chain := flow.Testnet.Chain()
+
+	t.Run("valid hex address", func(t *testing.T) {
+		t.Parallel()
+
+		validAddr := unittest.AddressFixture()
+		hexAddr := validAddr.Hex()
+
+		addr, err := convert.HexToAddress(hexAddr, chain)
+		require.NoError(t, err)
+		assert.Equal(t, validAddr, addr)
+	})
+
+	t.Run("empty hex address", func(t *testing.T) {
+		t.Parallel()
+
+		_, err := convert.HexToAddress("", chain)
+		require.Error(t, err)
+		assert.Equal(t, codes.InvalidArgument, status.Code(err))
+		assert.Contains(t, err.Error(), "address cannot be empty")
+	})
+
+	t.Run("invalid hex address for chain", func(t *testing.T) {
+		t.Parallel()
+
+		// Use an address from a different chain
+		mainnetAddr := unittest.RandomAddressFixtureForChain(flow.Mainnet)
+		hexAddr := mainnetAddr.Hex()
+
+		// This address should be invalid for Testnet chain
+		_, err := convert.HexToAddress(hexAddr, chain)
+		require.Error(t, err)
+		assert.Equal(t, codes.InvalidArgument, status.Code(err))
+		assert.Contains(t, err.Error(), "is invalid for chain")
+	})
+}
+
+// TestValidateBlockID tests the BlockID validation function.
+func TestValidateBlockID(t *testing.T) {
+	t.Parallel()
+
+	t.Run("valid block ID", func(t *testing.T) {
+		t.Parallel()
+
+		expectedID := unittest.IdentifierFixture()
+		rawID := expectedID[:]
+
+		blockID, err := convert.BlockID(rawID)
+		require.NoError(t, err)
+		assert.Equal(t, expectedID, blockID)
+	})
+
+	t.Run("invalid block ID length - too short", func(t *testing.T) {
+		t.Parallel()
+
+		invalidID := []byte{0x01, 0x02, 0x03}
+
+		_, err := convert.BlockID(invalidID)
+		require.Error(t, err)
+		assert.Equal(t, codes.InvalidArgument, status.Code(err))
+		assert.Contains(t, err.Error(), "invalid block id")
+	})
+
+	t.Run("invalid block ID length - too long", func(t *testing.T) {
+		t.Parallel()
+
+		invalidID := make([]byte, flow.IdentifierLen+1)
+
+		_, err := convert.BlockID(invalidID)
+		require.Error(t, err)
+		assert.Equal(t, codes.InvalidArgument, status.Code(err))
+		assert.Contains(t, err.Error(), "invalid block id")
+	})
+
+	t.Run("empty block ID", func(t *testing.T) {
+		t.Parallel()
+
+		_, err := convert.BlockID([]byte{})
+		require.Error(t, err)
+		assert.Equal(t, codes.InvalidArgument, status.Code(err))
+	})
+}
+
+// TestValidateBlockIDs tests the BlockIDs validation function.
+func TestValidateBlockIDs(t *testing.T) {
+	t.Parallel()
+
+	t.Run("valid random block IDs lists", func(t *testing.T) {
+		t.Parallel()
+
+		// Test multiple random lists to exercise conversion pathways
+		for count := 3; count < 8; count++ {
```

Comment:
I don't think we need to run this test multiple times as long as each test uses random data.